### PR TITLE
docs: adds clarification to imageset reference and examples

### PR DESCRIPTION
**Pull request by jpower432**
Signed-off-by: Jennifer Power <barnabei.jennifer@gmail.com>

# Description

This update imageset examples under `docs/examples` and updates the `imageset-config-ref.yaml` to provide valid catalog configuration examples.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue) (Docs)

# How Has This Been Tested?

N/A, docs PR

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/imageset-config-ref.yaml [hunk 1]
```diff
@@ -9,19 +9,25 @@ mirror:
...
+    graph: true # Include Cincinnati upgrade graph image in imageset (defaults to false)
   operators:
-    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.8 # References entire catalog
-      full: true # full can be set to pull a full catalog and must be set to filter packages
+    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.12 # References entire catalog
+      full: false # full set to false pull the latest version for all package channels with no versions set (default to false)
       packages:
+        - name: elasticsearch-operator
+          channels:
+            - name: 'stable-v0' # Mirrors a single channel
```
Comment (jharmison-redhat) [inline docs/imageset-config-ref.yaml:23]: Add clarity to comment

> Mirrors a single channel, from the previous channel head to the current channel head

File: docs/examples/imageset-config-filter-catalog.yaml [hunk 2]
```diff
@@ -1,33 +1,78 @@
...
+mirror:
+  operators:
+    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.10
+      packages:
+        - name: elasticsearch-operator
 ---
+# If a package is specified with a channel with no minVersion/maxVersion, only the channel's latest
+# version will be mirrored.
+
+# IMPORTANT: This is only true if metadata for this package in unestablished.
```
Comment (jharmison-redhat) [inline docs/examples/imageset-config-filter-catalog.yaml:19]: I think repetition in multiple examples is good, but multiple times in same file is excessive imo

File: docs/imageset-config-ref.yaml [hunk 3]
```diff
@@ -9,19 +9,25 @@ mirror:
...
         - name: rhacs-operator
-          minVersion: '3.67.0'
+          minVersion: '3.67.0' # Mirrors all version from the minimum version to the latest in the package
```
Comment (jharmison-redhat) [inline docs/imageset-config-ref.yaml:25]: Add clarity

> Mirrors all versions from the minimum version to the latest in the package, including all channels that contain any of those versions.

Consider addition of extra note - not here, but somewhere - that "minmum" and "latest" don't mean semver-style sorting, but rather relate to the graph constructed through 'replaces' and 'skips' in published versions

File: docs/examples/imageset-config-filter-catalog.yaml [hunk 4]
```diff
@@ -1,33 +1,78 @@
...
+mirror:
+  operators:
+    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.10
+      full: true
+      packages:
+        - name: elasticsearch-operator
+          channels:
+            - name: 'stable-v0'
+---
+#minVersion and maxVersion can be set that the channel level OR the package level. If set at the package level, all channels
```
Comment (jharmison-redhat) [inline docs/examples/imageset-config-filter-catalog.yaml:55]: can be set **at**

**Review by jharmison-redhat — COMMENTED** [review]
A couple of typos and a few spots that I think could use additional clarity.

**Review by sherine-k — APPROVED** [review]
This clarifies a lot. Thanks @jpower432 
/lgtm

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**